File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.cpp
class Solution {
public:
    int subtractProductAndSum(int n) {
        int product=1,sum=0;
        int temp=n;
        while(temp>0){
           int b=temp%10;
           sum+=b;
           temp/=10;
        }
        temp=n;
        while(temp>0){
            int c=temp%10;
            product*=c;
            temp/=10;
        }
        int diff=product-sum;
        return diff;
    }
};

File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.java
class Solution {
    public int product(int n){
        if(n==0){
            return 1;
        }
        return (n%10)*product(n/10);
    }
    public int sumdigits(int n){
        if(n==0){
            return 0;
        }
        return (n%10)+sumdigits(n/10);
    }
    public int subtractProductAndSum(int n) {
       int result1=product(n);
       int result2=sumdigits(n);
       return result1-result2;
        
    }
}